services/src/main/java/edu/unc/lib/dl/cdr/services/techmd/TechnicalMetadataEnhancement.java

```diff
@@ -131,45 +136,51 @@ public Element call() throws EnhancementException {
 			for (String dsid : ds2FitsDoc.keySet()) {
 				// get key PREMIS data
 				Document fits = ds2FitsDoc.get(dsid);
-				String md5checksum = fits.getRootElement().getChild("fileinfo", FITS_NS)
-						.getChildText("md5checksum", FITS_NS);
 				String size = fits.getRootElement().getChild("fileinfo", FITS_NS).getChildText("size", FITS_NS);
-
-				// IDENTIFICATION LOGIC
-				// get mimetype out of FITS XML
-				Element trustedIdentity = null;
-				Element idn = fits.getRootElement().getChild("identification", ns);
-				for (Object child : idn.getChildren("identity", ns)) {
-					Element el = (Element) child;
-					if (idn.getAttributeValue("status") == null
-							|| el.getChildren("tool", ns).size() > 1
-							|| (!"Exiftool".equals(el.getChild("tool", ns).getAttributeValue("toolname")) && !"application/x-symlink"
-									.equals(el.getAttributeValue("mimetype")))) {
-						trustedIdentity = el;
-						break;
-					}
-				}
-
-				String fitsMimetype = null;
+				
+				String fedoraMimetype = sourceMimetype.get(dsid);
+				String fitsMimetype = fedoraMimetype;
+				// Format only provided if using FITS mimetype, since otherwise it may not match
 				String format = null;
-				if (trustedIdentity != null) {
-					fitsMimetype = trustedIdentity.getAttributeValue("mimetype");
-					format = trustedIdentity.getAttributeValue("format");
-				} else {
-					format = "Unknown";
-					LOG.warn("FITS unable to conclusively identify file: {}/{}", pid, dsid);
-					LOG.info(new XMLOutputter().outputString(fits));
-				}
+				
+				// If fedora mimetype is not meaningful, then override with FITS generated value
+				if (fedoraMimetype == null || fedoraMimetype.trim().length() == 0
+						|| fedoraMimetype.contains("octet-stream")) {
+					
+					// IDENTIFICATION LOGIC
+					// get mimetype out of FITS XML
+					Element identity = null;
+					Element idn = fits.getRootElement().getChild("identification", ns);
```

Comment:
True, the status test every loop through is weird.  The xpath looks like it'd work, although i haven't tested it.